### DISCUSSION TITLE
Login interface with only ID field at first

### DISCUSSION THREAD

**Discussion by LasseRosenow** [ory/kratos#3010] [Q&A]
Many famous identity providers these days implement login in a way that only asks for the E-Mail at first and then asks for either password, 2fa etc. depending on which login method is connected to that e-mail.

I think this would improve the login experience for the user a lot.

The current implementation, that shows multiple login methods just below each other is confusing, as most users don't know, what all these input fields are for.

So my question is, if the kratos api currently allows to build a login-ui that implements this kind of user-flow?

And if not: Is this planned?

**Comment by vinckr** [comment]
Hello @LasseRosenow 
This is possible right now, as you have total control over the UI. You can build your own UI in whatever way you want. 

> I think this would improve the login experience for the user a lot.

Personally I dislike this kind of flow and prefer to just have one page with maybe 2 social login options and username/pw. It also messes with password managers (most of the time it works but sometimes not).  
I get that this is a personal preference, I was just wondering where you see the advantage? 
Having username/email and password field is quite straightforward, or what kind of login UI are you considering? 

Let me know how it goes implementing this, it would make a good guide for others as well!

**Reply by LasseRosenow** [reply]
Having username and password in combination with social login is also something that I think is fine, but there are other login methods that require a username but no password.

If I enabled those password less flows like Passkeys or email verification, or sms etc.
It makes sense not to show the password, but just the e-mail and then ask for what is needed for this specific account. Either a password or a e-mail code verification, a pin of a hardware key etc.

Right now if you enable password less with ory, you get an additional "email, sign in with secure key" card below the "email, password" section, so it feels very duplicated.

> It also messes with password managers (most of the time it works but sometimes not).

It works if the password field is part of the DOM, but input type hidden, Or opacity. 0, height 0, not sure, but this can be implemented to work :) of course I would take care of it :)

> This is possible right now, as you have total control over the UI. You can build your own UI in whatever way you want.

But the problem is, that I need to find out, what authentication method the given e-mail requires. So my true question is, if kratos has an API that can give me this information, so I can either show a password field, or ask for an e-mail code etc. After the Username/email was provided, depending on what is needed.

**Reply by vinckr** [reply]
Gotcha, 
You should get all available authN methods in the `methods` payload in the [Login Flow](https://www.ory.sh/docs/kratos/self-service/flows/user-login).

You can also get the identity, but then you need the identities ID, not sure what the best way is to associate it with the email/username 🤔 (we have it on the roadmap to improve this btw so you can get the identity just from the identifier) 
https://www.ory.sh/docs/reference/api#tag/identity/operation/getIdentity

Let me know if that helps and how you proceed further :-)

**Reply by LasseRosenow** [reply]
Thank you very much, this is looking very promising :) I let you know, how we proceeded in implementing it for ourselves :)

**Reply by LasseRosenow** [reply]
using the `getIdentity` method only works from a admin api context, so we have to write a service in between exposing this functionality to our frontend client. 

Is getting the identity just from the identifier as part of your roadmap meant to expose this as a non-admin API?

**Reply by vinckr** [reply]
I am not sure if it will be available as non-admin API. 
My educated guess is it will not change the current getIdentity API too much, just that you can supply also an email instead of the ID. So it will most likely remain an admin API. 
I will have a look if I can find an issue that tracks this feature.

**Reply by thomastthai** [reply]
> using the `getIdentity` method only works from a admin api context, so we have to write a service in between exposing this functionality to our frontend client.

@LasseRosenow, I'm heading down a similar path as you and would love to see your final implementation. The login form would have just the email address field shown with a submit button. The other fields or options would only show on the same login page after available authN methods are determined.

**Reply by LasseRosenow** [reply]
> @LasseRosenow, I'm heading down a similar path as you and would love to see your final implementation. The login form would have just the email address field shown with a submit button. The other fields or options would only show on the same login page after available authN methods are determined. 

Yes this is the same approach as ours. We will start implementing it somewhere around the next few weeks... I will share our solution here :)
